src/transformers/modeling_tf_bert.py

```diff
@@ -921,76 +1199,116 @@ def call(
             return_dict=return_dict,
             training=training,
         )
-
         sequence_output = outputs[0]
-        prediction_scores = self.mlm(sequence_output, training=training)
-        loss = None if labels is None else self.compute_loss(labels, prediction_scores)
+        prediction_scores = self.cls(sequence_output=sequence_output, training=training)
+        lm_loss = None
 
-        if not return_dict:
-            output = (prediction_scores,) + outputs[2:]
-            return ((loss,) + output) if loss is not None else output
+        if labels is not None:
+            # shift labels to the left and cut last logit token
+            shifted_prediction_scores = prediction_scores[:, :-1]
+            labels = labels[:, 1:]
+            lm_loss = self.compute_loss(labels=labels, logits=shifted_prediction_scores)
 
-        return TFMaskedLMOutput(
-            loss=loss,
+        if tf.executing_eagerly():
+            if not return_dict:
+                output = (prediction_scores,) + outputs[2:]
+
+                return ((prediction_scores,) + output) if lm_loss is not None else output
+
+        return TFCausalLMOutput(
+            loss=lm_loss,
             logits=prediction_scores,
             hidden_states=outputs.hidden_states,
             attentions=outputs.attentions,
         )
 
 
-class TFBertLMHeadModel(TFBertPreTrainedModel, TFCausalLanguageModelingLoss):
-
-    authorized_unexpected_keys = [r"pooler"]
-    authorized_missing_keys = [r"pooler"]
+@add_start_docstrings("""Bert Model with a `language modeling` head on top. """, BERT_START_DOCSTRING)
+class TFBertForMaskedLM(TFBertPreTrainedModel, TFMaskedLanguageModelingLoss):
+    authorized_unexpected_keys = [r"pooler", r"seq_relationship"]
 
     def __init__(self, config, *inputs, **kwargs):
         super().__init__(config, *inputs, **kwargs)
 
-        if not config.is_decoder:
-            logger.warning("If you want to use `TFBertLMHeadModel` as a standalone, add `is_decoder=True.`")
+        if config.is_decoder:
+            logger.warning(
+                "If you want to use `TFBertForMaskedLM` make sure `config.is_decoder=False` for "
+                "bi-directional self-attention."
+            )
 
-        self.bert = TFBertMainLayer(config, name="bert")
-        self.mlm = TFBertMLMHead(config, self.bert.embeddings, name="mlm___cls")
+        self.initializer_range = config.initializer_range
+        self.bert = TFBertMainLayer(config=config, add_pooling_layer=False, name="bert")
+        self.cls = TFBertOnlyMLMHead(config=config, input_embeddings=self.bert.embeddings, name="cls")
```

Comment:
We should not rename parameter weights I think `mlm__cls` => `cls` means that all previously trained models cannot be loaded anymore, no?